src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/StackSpiller.cs

```diff
@@ -1089,15 +1089,15 @@ private static void RequireNoRefArgs(MethodBase method)
         /// <remarks>
         /// We could support this if spilling happened later in the compiler.
         /// </remarks>
-        private static void RequireNotRefInstance(Expression instance)
+        private static void RequireNotRefInstance(Expression? instance)
         {
             if (IsRefInstance(instance))
             {
-                throw Error.TryNotSupportedForValueTypeInstances(instance.Type);
+                throw Error.TryNotSupportedForValueTypeInstances(instance!.Type);
             }
         }
 
-        private static bool IsRefInstance(Expression instance)
+        private static bool IsRefInstance(Expression? instance)
```

Comment:
>Expression? [](start = 42, length = 11)

Consider adding `[NotNullWhen(true)]`. Then it should be possible to remove the `!` in the previous method.